app/src/main/java/org/hisp/dhis/android/dataentry/form/section/viewmodels/FormItemViewModelFactoryImpl.java

```diff
@@ -0,0 +1,205 @@
+package org.hisp.dhis.android.dataentry.form.section.viewmodels;
+
+import android.database.Cursor;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.InputFilter;
+import android.text.InputType;
+
+import org.hisp.dhis.android.core.common.ValueType;
+import org.hisp.dhis.android.dataentry.R;
+import org.hisp.dhis.android.dataentry.commons.utils.Preconditions;
+import org.hisp.dhis.android.dataentry.form.section.EditTextHintCache;
+import org.hisp.dhis.android.dataentry.form.section.viewmodels.checkbox.CheckBoxViewModel;
+import org.hisp.dhis.android.dataentry.form.section.viewmodels.coordinate.CoordinateViewModel;
+import org.hisp.dhis.android.dataentry.form.section.viewmodels.date.DateViewModel;
+import org.hisp.dhis.android.dataentry.form.section.viewmodels.edittext.EditTextViewModel;
+import org.hisp.dhis.android.dataentry.form.section.viewmodels.optionset.OptionSetViewModel;
+import org.hisp.dhis.android.dataentry.form.section.viewmodels.radiobutton.RadioButtonViewModel;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import static android.text.InputType.TYPE_CLASS_TEXT;
+
+final public class FormItemViewModelFactoryImpl implements FormItemViewModelFactory {
```

Comment:
The problem is that this class doesn't have one responsibility. It is responsible for instantiating around 10 view models, where each model is created in own way. It also doesn't make much sense to have factory methods for different models in one place (especially that big).